### ISSUE TITLE
consumes annotation field on HTTP actions without a body

### ISSUE THREAD

**Issue by kkrauth** [zalando-stups/swagger-codegen-tooling#45]
When using a swagger file that defines the consume/produce properties globally:
```
consumes:
  - application/json
produces:
  - application/json
```
In the resulting spring interface definitions, the consumes request mapping field is included even for HTTP actions that don't have a body (i.e. DELETE):
```
@RequestMapping(value="/myendpoint",method=RequestMethod.DELETE,consumes={ "application/json" } , produces={ "application/json" })
```
This causes issues with "strict" HTTP client libraries like OkHttp which will refuse to send a Content-Type header for DELETE/GET operations - as a result, spring will reject these requests because of an unsupported content type. Would you consider omitting the consumes type for GET/DELETE?

It could be argued that perhaps spring should ignore the consumes request mapping field for HTTP verbs that don't contain request bodies.

**Comment by jbellmann** [comment]
@kkrauth 

this was reported [here](https://github.com/zalando-stups/swagger-codegen-tooling/issues/41). But the issues is not closed yet because the only method fixed so far should be GET. Could you please verify that GET-methods are generated correctly?
Do you have more methods in mind where "consumes" makes no sense?

Regards
Jörg

**Comment by kkrauth** [comment]
Thanks, didn't spot that issue. I think GET and DELETE are the two most commonly used HTTP methods that shouldn't have a request body. So if those two omit the content type attribute, the majority of use cases should be covered.

I will add a GET endpoint and check if it works correct with 0.4.31.

**Comment by jbellmann** [comment]
I released 0.4.33 minutes ago. DELETE-method should not be generated with 'consumes' anymore.
Please verify and if it's working as expected close this issue. Otherwise let me know what went wrong.

Thanks
Jörg

**Comment by kkrauth** [comment]
Yep, that's good! Thank you.
